skimage/segmentation/_watershed.py

```diff
@@ -119,7 +119,7 @@ def watershed(image, markers=None, connectivity=1, offset=None, mask=None,
     Returns
     -------
     out : ndarray
-        A labeled matrix of the same type and shape as markers
+        A labeled matrix of dtype `int32` and the shape as markers
```

Comment:
Ideally, we should change the function so that the same dtype as the markers is returned, rather than the documentation. It shouldn't be too hard a change.

Reply:
Hi @jni , When I test my approach locally, I get the following error after changing every instance of `'DTYPE_INT32_t'` to `np_anyint` in the `_watershed_cy.pyx` file. 

`ValueError: Buffer dtype mismatch, expected 'DTYPE_INT32_t' but got 'unsigned short'`

All test passed okay. Any suggestions ?

Reply:
Are you sure you've changed *every* instance? It sounds from the error like you missed one. (There is more than one relevant file, also heap_watershed, for example) If you push your changes I can have a look.

The tests pass probably because the compilation failed but the old compiled functions are still there.

Reply:
I did change all the instances in my previous attempts. I figured I was testing my changes wrong hence the `ValueError`. Eureka

File compiled successfully and test passed. 
I have pushed my changes. Please review.
Thank you.